From 10e6b63b8d1f3b0579f7333f94491ac05aca10fb Mon Sep 17 00:00:00 2001
From: Alexandre Belloni <alexandre.belloni@free-electrons.com>
Date: Mon, 29 Apr 2013 18:14:08 +0200
Subject: [PATCH 078/138] Input: atmel_tsadcc: fix compilation as a module

Signed-off-by: Alexandre Belloni <alexandre.belloni@free-electrons.com>
Signed-off-by: Nicolas Ferre <nicolas.ferre@atmel.com>
---
 drivers/input/touchscreen/atmel_tsadcc.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/touchscreen/atmel_tsadcc.c b/drivers/input/touchscreen/atmel_tsadcc.c
index 041c85b..5fc52c6 100644
--- a/drivers/input/touchscreen/atmel_tsadcc.c
+++ b/drivers/input/touchscreen/atmel_tsadcc.c
@@ -446,7 +446,7 @@ static int atmel_tsadcc_remove(struct platform_device *pdev)
 static const struct of_device_id atmel_tsaddcc_dt_ids[] = {
 	{ .compatible = "atmel,at91sam9x5-tsadcc"},
 	{ /* sentinel */ }
-}
+};
 
 MODULE_DEVICE_TABLE(of, atmel_tsaddcc_dt_ids);
 #endif
-- 
1.7.10.4

