From 7f0d71524e593f1022a04968b41ec4801a791fac Mon Sep 17 00:00:00 2001
From: Ludovic Desroches <ludovic.desroches@atmel.com>
Date: Tue, 16 Oct 2012 17:08:54 +0200
Subject: [PATCH 131/138] video: atmel_lcdfb: add pinctrl

Signed-off-by: Ludovic Desroches <ludovic.desroches@atmel.com>
---
 drivers/video/atmel_lcdfb_core.c |    9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/video/atmel_lcdfb_core.c b/drivers/video/atmel_lcdfb_core.c
index 13f2573..9ccd408 100644
--- a/drivers/video/atmel_lcdfb_core.c
+++ b/drivers/video/atmel_lcdfb_core.c
@@ -11,6 +11,7 @@
 #include <linux/kernel.h>
 #include <linux/module.h>
 #include <linux/platform_device.h>
+#include <linux/pinctrl/consumer.h>
 #include <linux/dma-mapping.h>
 #include <linux/interrupt.h>
 #include <linux/clk.h>
@@ -555,6 +556,7 @@ int __atmel_lcdfb_probe(struct platform_device *pdev,
 	struct resource *regs = NULL, *clut = NULL;
 	struct resource *map = NULL;
 	int ret;
+	struct pinctrl *pinctrl;
 
 	dev_dbg(dev, "%s BEGIN\n", __func__);
 
@@ -634,6 +636,13 @@ int __atmel_lcdfb_probe(struct platform_device *pdev,
 		goto stop_clk;
 	}
 
+	pinctrl = devm_pinctrl_get_select_default(dev);
+	if (IS_ERR(pinctrl)) {
+		dev_err(dev, "Failed to request pinctrl\n");
+		ret = PTR_ERR(pinctrl);
+		goto stop_clk;
+	}
+
 	/* No error checking, some devices can do without IRQ */
 	sinfo->irq_base = platform_get_irq(pdev, 0);
 
-- 
1.7.10.4

