From 45ec305965fa93bffe01c6a530e75c7eeb11b35e Mon Sep 17 00:00:00 2001
From: Bo Shen <voice.shen@atmel.com>
Date: Wed, 28 Nov 2012 17:21:43 +0800
Subject: [PATCH 132/319] input: qt1070: add irq pinctrl consumer

Signed-off-by: Bo Shen <voice.shen@atmel.com>
---
 drivers/input/keyboard/qt1070.c |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/input/keyboard/qt1070.c b/drivers/input/keyboard/qt1070.c
index ca68f29..b266dbf 100644
--- a/drivers/input/keyboard/qt1070.c
+++ b/drivers/input/keyboard/qt1070.c
@@ -33,6 +33,7 @@
 #include <linux/interrupt.h>
 #include <linux/jiffies.h>
 #include <linux/delay.h>
+#include <linux/pinctrl/consumer.h>
 
 /* Address for each register */
 #define CHIP_ID            0x00
@@ -145,9 +146,16 @@ static int __devinit qt1070_probe(struct i2c_client *client,
 {
 	struct qt1070_data *data;
 	struct input_dev *input;
+	struct pinctrl *pinctrl;
 	int i;
 	int err;
 
+	pinctrl = devm_pinctrl_get_select_default(&client->dev);
+	if (IS_ERR(pinctrl)) {
+		dev_err(&client->dev, "Failed to request pinctrl\n");
+		return PTR_ERR(pinctrl);
+	}
+
 	err = i2c_check_functionality(client->adapter, I2C_FUNC_SMBUS_BYTE);
 	if (!err) {
 		dev_err(&client->dev, "%s adapter not supported\n",
-- 
1.7.10.4

