From 8d86f812c1e3015c8c9c1ce92bf0f67a2124c0fc Mon Sep 17 00:00:00 2001
From: Nicolas Ferre <nicolas.ferre@atmel.com>
Date: Thu, 16 May 2013 12:36:44 +0200
Subject: [PATCH 137/138] Input: atmel_tsadcc: remove __devinit

Signed-off-by: Nicolas Ferre <nicolas.ferre@atmel.com>
---
 drivers/input/touchscreen/atmel_tsadcc.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/touchscreen/atmel_tsadcc.c b/drivers/input/touchscreen/atmel_tsadcc.c
index 5fc52c6..89eb176 100644
--- a/drivers/input/touchscreen/atmel_tsadcc.c
+++ b/drivers/input/touchscreen/atmel_tsadcc.c
@@ -223,7 +223,7 @@ static int atmel_of_init_tsadcc(struct device_node *np,
 	return 0;
 }
 #else
-static int __devinit atmel_of_init_tsadcc(struct device_node *np,
+static int atmel_of_init_tsadcc(struct device_node *np,
 				struct at91_tsadcc_data *pdata,
 				struct platform_device *pdev)
 {
-- 
1.7.10.4

