From 20e47ad6313a4e5219bf2b97c30f0c5f2554390d Mon Sep 17 00:00:00 2001
From: Josh Wu <josh.wu@atmel.com>
Date: Wed, 14 Nov 2012 17:57:51 +0800
Subject: [PATCH 242/319] atmel_nand: nfc:  add interrupt definitions.

---
 drivers/mtd/nand/atmel_nand_nfc.h |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/mtd/nand/atmel_nand_nfc.h b/drivers/mtd/nand/atmel_nand_nfc.h
index 46eb111..fc745af 100644
--- a/drivers/mtd/nand/atmel_nand_nfc.h
+++ b/drivers/mtd/nand/atmel_nand_nfc.h
@@ -53,6 +53,9 @@
 #define		ATMEL_HSMC_NFC_ASE		(1 << 23)
 #define		ATMEL_HSMC_NFC_RB_EDGE		(1 << 24)
 
+#define ATMEL_HSMC_NFC_IER		0x0c	/* NFC Interrupt Enable Register */
+#define ATMEL_HSMC_NFC_IDR		0x10	/* NFC Interrupt Disable Register */
+#define ATMEL_HSMC_NFC_IMR		0x14	/* NFC Interrupt Mask Register */
 #define ATMEL_HSMC_NFC_CYCLE0		0x18	/* NFC Address Cycle Zero Register */
 #define		ATMEL_HSMC_NFC_ADDR_CYCLE0	(0xff)
 
-- 
1.7.10.4

