From 3206169901826a4f90d0a58bc8a51ff50916038f Mon Sep 17 00:00:00 2001
From: Nicolas Ferre <nicolas.ferre@atmel.com>
Date: Thu, 20 Dec 2012 12:04:53 +0100
Subject: [PATCH 263/319] ARM: at91/board-dt: add fake ISI material

Signed-off-by: Nicolas Ferre <nicolas.ferre@atmel.com>
---
 arch/arm/mach-at91/board-dt.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/mach-at91/board-dt.c b/arch/arm/mach-at91/board-dt.c
index 5427f86..0dda409 100644
--- a/arch/arm/mach-at91/board-dt.c
+++ b/arch/arm/mach-at91/board-dt.c
@@ -271,6 +271,12 @@ static struct platform_device isi_ov2640 = {
 static struct platform_device *devices[] __initdata = {
 	&isi_ov2640,
 };
+#else
+static struct soc_camera_link iclink_ov2640;
+static int i2c_camera_power_revB(struct device *dev, int on)
+{
+	return -1;
+}
 #endif
 
 struct of_dev_auxdata at91_auxdata_lookup[] __initdata = {
-- 
1.7.10.4

